connector/solarwindsentityconnector/internal/event.go

```diff
@@ -15,47 +15,135 @@
 package internal
 
 import (
+	"fmt"
+	"time"
+
 	"github.com/solarwinds/solarwinds-otel-collector-releases/connector/solarwindsentityconnector/config"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.uber.org/zap"
-	"time"
 )
 
-func AppendEntityUpdateEvent(lrs *plog.LogRecordSlice, entity config.Entity, resourceAttrs pcommon.Map) {
-	event, err := CreateEntityEvent(resourceAttrs, entity)
+type EventBuilder struct {
+	Entities       map[string]config.Entity
+	Relationships  []config.Relationship
+	SourcePrefix   string
+	DestPrefix     string
+	ResultEventLog *plog.LogRecordSlice
+	Logger         *zap.Logger
+}
+
+func NewEventBuilder(entities map[string]config.Entity, relationships []config.Relationship, sourcePrefix string, destPrefix string, events *plog.Logs, logger *zap.Logger) *EventBuilder {
+	return &EventBuilder{
+		Entities:       entities,
+		Relationships:  relationships,
+		SourcePrefix:   sourcePrefix,
+		DestPrefix:     destPrefix,
+		ResultEventLog: CreateResultEventLog(events),
+		Logger:         logger,
+	}
+}
+
+// CreateResultEventLog prepares a clean LogRecordSlice, where log records representing events should be appended.
+// In given plog.Logs creates a resource log with one scope log and set attributes needed by SWO ingestion.
+func CreateResultEventLog(logs *plog.Logs) *plog.LogRecordSlice {
```

Comment:
```suggestion
func createResultEventLog(logs *plog.Logs) *plog.LogRecordSlice {
```
this is used only when creating a new Event builder.

Reply:
fixed